use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,491 @@
+You must update the configuration.toml for Kata Containers to point to the 
+custom kernel, custom rootfs, and to specify which modules to load when the 
+virtual machine is booted when a container is run. The example below assumes 
+an Intel QAT is installed and those modules need to be loaded.
+
+```
+$ sudo mkdir -p /etc/kata-containers
+$ sudo cp /usr/share/defaults/kata-containers/configuration-qemu.toml /etc/kata-containers/configuration.toml
+
+# Make sure usdm_drv is loaded in the image
+$ sudo sed -i "s|kernel_params = \"\"|kernel_params = \"modules-load=usdm_drv,qat_c62xvf\"|g" /etc/kata-containers/configuration.toml
+
+# Change the default kernel and rootfs
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/kata-containers.img|${KATA_KERNEL_LOCATION}\/kata-containers.img|g" /etc/kata-containers/configuration.toml
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/vmlinuz.container|${KATA_ROOTFS_LOCATION}\/vmlinuz-${LINUX_VER}_qat|g" /etc/kata-containers/configuration.toml
+```
+
+## Verify Intel QAT works in a Docker Kata Containers container
+
+The following instructions leverage an OpenSSL DockerFile that builds the 
+Intel QAT engine to allow OpenSSL to offload crypto functions. It is a 
+convenient way to test that VFIO device passthrough for the Intel QAT VF’s are
+working properly with the Kata Containers VM.
+
+## Build OpenSSL Intel QAT engine container
+
+There is an OpenSSL Intel QAT Dockerfile located at 
+[https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine). 
+Use this file to build a container image with an optimized OpenSSL engine for 
+Intel QAT. Using Docker build with the kata runtime can sometimes have issues. 
+Therefore, it is recommended to change the default runtime to runc before 
+doing a build. Instructions for this are below.
+
+__Note:__ The Intel QAT driver version in this container might not match the 
+Intel QAT driver compiled and loaded on the host.
+
+```
+$ cd $QAT_SRC
+$ curl -O $QAT_DOCKERFILE
+$ sudo sed -i 's/kata-runtime/runc/g' /etc/systemd/system/docker.service.d/50-runtime.conf
+$ sudo systemctl daemon-reload && sudo systemctl restart docker
+$ docker build -t openssl-qat-engine .
+```
+
+### Test Intel QAT in Docker
+
+The host should already be setup with 16 virtual functions of the Intel QAT 
+card bound to VFIO-PCI. This can be verified by looking in `/dev/VFIO` for a 
+listing of functions. Replace the number 90 with one of the VF’s exposed 
+in `/dev/VFIO`. It might be required to add an `IPC_LOCK` capability to your 
+Docker runtime depending on which rootfs is used. 
+
+```
+$ docker run -it --runtime=kata-runtime --cap-add=IPC_LOCK --device=/dev/VFIO/90 -v /dev:/dev -v ${QAT_CONF_LOCATION}:/etc openssl-qat-engine bash
+```
+
+Below are some commands to run in the container image to verify Intel QAT is 
+working
+
+```
+bash-5.0# cat /proc/modules
```

Comment:
I guess you got the `bash-5.0#` prefix from a copy/paste from the docker image prompt? Not sure if we should keep it like that, or use the standard kata docs format of ```$ cmd_goes_here``` ?

Reply:
I pinged Kristi and copied you to get clarity on the best way to show this.